operator/pkg/manifest/shared_test.go

```diff
@@ -96,3 +100,104 @@ func TestReadLayeredYAMLs(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveConflictSettings(t *testing.T) {
+	getIopFromStr := func(y string) *iopv1alpha1.IstioOperator {
+		var iop iopv1alpha1.IstioOperator
+		json.Unmarshal([]byte(y), &iop)
+		return &iop
+	}
+
+	tests := []struct {
+		name     string
+		givenIop *iopv1alpha1.IstioOperator
+		wantIop  *iopv1alpha1.IstioOperator
+	}{
+		{
+			name: "minAvailable & maxUnavailable both set to 0",
+			givenIop: getIopFromStr(`{"apiVersion": "install.istio.io/v1alpha1",
+				 "kind": "IstioOperator",
+				 "metadata": { "name": "istio", "namespace": "istio-operator"},
+				 "spec": {"components": {"pilot": {"k8s": {"podDisruptionBudget": {
+            	 "maxUnavailable": 0, "minAvailable": 0}}}}}}`),
+			wantIop: getIopFromStr(`{"apiVersion": "install.istio.io/v1alpha1",
+			     "kind": "IstioOperator",
+			     "metadata": { "name": "istio", "namespace": "istio-operator"},
+			     "spec": {"components": {"pilot": {"k8s": {"podDisruptionBudget": {
+				 "maxUnavailable": 0, "minAvailable": 1}}}}}}`),
```

Comment:
this is hard to read in json and not the convention for istio tests. could you please rewrite as YAML (you can use python, an online tool, etc. to easily convert)

Reply:
@ostromart will change that to using yaml or directly create iop object.

Reply:
changed yaml presentation of the test data in test file, please take a look, thanks.